### PR TITLE
[SPARK-30889][SPARK-30913][CORE][DOC] Add version information to the configuration of Tests.scala and Worker

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
1.Add version information to the configuration of `Tests` and `Worker`.
2.Update the docs of `Worker`.

I sorted out some information of `Tests` show below.

Item name | Since version | JIRA ID | Commit ID | Note
-- | -- | -- | -- | --
spark.testing.memory | 1.6.0 | SPARK-10983 | b3ffac5178795f2d8e7908b3e77e8e89f50b5f6f#diff-395d07dcd46359cca610ce74357f0bb4 |  
spark.testing.dynamicAllocation.scheduleInterval | 2.3.0 | SPARK-22864 | 4e9e6aee44bb2ddb41b567d659358b22fd824222#diff-b096353602813e47074ace09a3890d56 |  
spark.testing | 1.0.1 | SPARK-1606 | ce57624b8232159fe3ec6db228afc622133df591#diff-d239aee594001f8391676e1047a0381e |  
spark.test.noStageRetry | 1.2.0 | SPARK-3796 | f55218aeb1e9d638df6229b36a59a15ce5363482#diff-6a9ff7fb74fd490a50462d45db2d5e11 |  
spark.testing.reservedMemory | 1.6.0 | SPARK-12081 | 84c44b500b5c90dffbe1a6b0aa86f01699b09b96#diff-395d07dcd46359cca610ce74357f0bb4 | 
spark.testing.nHosts | 3.0.0 | SPARK-26491 | 1a641525e60039cc6b10816e946cb6f44b3e2696#diff-8b4ea8f3b0cc1e7ce7e943de1abbb165 |  
spark.testing.nExecutorsPerHost | 3.0.0 | SPARK-26491 | 1a641525e60039cc6b10816e946cb6f44b3e2696#diff-8b4ea8f3b0cc1e7ce7e943de1abbb165 |  
spark.testing.nCoresPerExecutor | 3.0.0 | SPARK-26491 | 1a641525e60039cc6b10816e946cb6f44b3e2696#diff-8b4ea8f3b0cc1e7ce7e943de1abbb165 |  
spark.resources.warnings.testing | 3.1.0 | SPARK-29148 | 496f6ac86001d284cbfb7488a63dd3a168919c0f#diff-8b4ea8f3b0cc1e7ce7e943de1abbb165 |  
spark.testing.resourceProfileManager | 3.1.0 | SPARK-29148 | 496f6ac86001d284cbfb7488a63dd3a168919c0f#diff-8b4ea8f3b0cc1e7ce7e943de1abbb165 |  

I sorted out some information of `Worker` show below.

Item name | Since version | JIRA ID | Commit ID | Note
-- | -- | -- | -- | --
spark.worker.resourcesFile | 3.0.0 | SPARK-27369 | 7cbe01e8efc3f6cd3a0cac4bcfadea8fcc74a955#diff-b2fc8d6ab7ac5735085e2d6cfacb95da |  
spark.worker.timeout | 0.6.2 | None | e395aa295aeec6767df798bf1002b1f30983c1cd#diff-776a630ac2b2ec5fe85c07ca20a58fc0 |  
spark.worker.driverTerminateTimeout | 2.1.2 | SPARK-20843 | ebd72f453aa0b4f68760d28b3e93e6dd33856659#diff-829a8674171f92acd61007bedb1bfa4f |  
spark.worker.cleanup.enabled | 1.0.0 | SPARK-1154 | 1440154c27ca48b5a75103eccc9057286d3f6ca8#diff-916ca56b663f178f302c265b7ef38499 |  
spark.worker.cleanup.interval | 1.0.0 | SPARK-1154 | 1440154c27ca48b5a75103eccc9057286d3f6ca8#diff-916ca56b663f178f302c265b7ef38499 |  
spark.worker.cleanup.appDataTtl | 1.0.0 | SPARK-1154 | 1440154c27ca48b5a75103eccc9057286d3f6ca8#diff-916ca56b663f178f302c265b7ef38499 |  
spark.worker.preferConfiguredMasterAddress | 2.2.1 | SPARK-20529 | 75e5ea294c15ecfb7366ae15dce196aa92c87ca4#diff-916ca56b663f178f302c265b7ef38499 |  
spark.worker.ui.port | 1.1.0 | SPARK-2857 | 12f99cf5f88faf94d9dbfe85cb72d0010a3a25ac#diff-48ca297b6536cb92362bec1487581f05 |  
spark.worker.ui.retainedExecutors | 1.5.0 | SPARK-9202 | c0686668ae6a92b6bb4801a55c3b78aedbee816a#diff-916ca56b663f178f302c265b7ef38499 | 
spark.worker.ui.retainedDrivers | 1.5.0 | SPARK-9202 | c0686668ae6a92b6bb4801a55c3b78aedbee816a#diff-916ca56b663f178f302c265b7ef38499 | 
spark.worker.ui.compressedLogFileLengthCacheSize | 2.0.2 | SPARK-17711 | 26e978a93f029e1a1b5c7524d0b52c8141b70997#diff-d239aee594001f8391676e1047a0381e |  
spark.worker.decommission.enabled | 3.1.0 | SPARK-20628 | d273a2bb0fac452a97f5670edd69d3e452e3e57e#diff-b2fc8d6ab7ac5735085e2d6cfacb95da |  

### Why are the changes needed?
Supplemental configuration version information.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Exists UT

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/internal/config/Tests.scala [hunk 1]
```diff
@@ -22,43 +22,54 @@ private[spark] object Tests {
   val TEST_USE_COMPRESSED_OOPS_KEY = "spark.test.useCompressedOops"
 
   val TEST_MEMORY = ConfigBuilder("spark.testing.memory")
+    .version("1.6.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Tests.scala]: SPARK-10983, commit ID: b3ffac5178795f2d8e7908b3e77e8e89f50b5f6f#diff-395d07dcd46359cca610ce74357f0bb4

File: core/src/main/scala/org/apache/spark/internal/config/Tests.scala [hunk 2]
```diff
@@ -22,43 +22,54 @@ private[spark] object Tests {
...
     .longConf
     .createWithDefault(Runtime.getRuntime.maxMemory)
 
   val TEST_SCHEDULE_INTERVAL =
     ConfigBuilder("spark.testing.dynamicAllocation.scheduleInterval")
+      .version("2.3.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Tests.scala]: SPARK-22864, commit ID: 4e9e6aee44bb2ddb41b567d659358b22fd824222#diff-b096353602813e47074ace09a3890d56

File: core/src/main/scala/org/apache/spark/internal/config/Tests.scala [hunk 3]
```diff
@@ -22,43 +22,54 @@ private[spark] object Tests {
...
       .longConf
       .createWithDefault(100)
 
   val IS_TESTING = ConfigBuilder("spark.testing")
+    .version("1.0.1")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Tests.scala]: SPARK-1606, commit ID: ce57624b8232159fe3ec6db228afc622133df591#diff-d239aee594001f8391676e1047a0381e

File: core/src/main/scala/org/apache/spark/internal/config/Tests.scala [hunk 4]
```diff
@@ -22,43 +22,54 @@ private[spark] object Tests {
...
     .booleanConf
     .createOptional
 
   val TEST_NO_STAGE_RETRY = ConfigBuilder("spark.test.noStageRetry")
+    .version("1.2.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Tests.scala]: SPARK-3796, commit ID: f55218aeb1e9d638df6229b36a59a15ce5363482#diff-6a9ff7fb74fd490a50462d45db2d5e11

File: core/src/main/scala/org/apache/spark/internal/config/Tests.scala [hunk 5]
```diff
@@ -22,43 +22,54 @@ private[spark] object Tests {
...
     .booleanConf
     .createWithDefault(false)
 
   val TEST_RESERVED_MEMORY = ConfigBuilder("spark.testing.reservedMemory")
+    .version("1.6.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Tests.scala]: SPARK-12081, commit ID: 84c44b500b5c90dffbe1a6b0aa86f01699b09b96#diff-395d07dcd46359cca610ce74357f0bb4

File: core/src/main/scala/org/apache/spark/internal/config/Tests.scala [hunk 6]
```diff
@@ -22,43 +22,54 @@ private[spark] object Tests {
...
     .longConf
     .createOptional
 
   val TEST_N_HOSTS = ConfigBuilder("spark.testing.nHosts")
+    .version("3.0.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Tests.scala]: SPARK-26491, commit ID: 1a641525e60039cc6b10816e946cb6f44b3e2696#diff-8b4ea8f3b0cc1e7ce7e943de1abbb165

File: core/src/main/scala/org/apache/spark/internal/config/Tests.scala [hunk 7]
```diff
@@ -22,43 +22,54 @@ private[spark] object Tests {
...
     .intConf
     .createWithDefault(5)
 
   val TEST_N_EXECUTORS_HOST = ConfigBuilder("spark.testing.nExecutorsPerHost")
+    .version("3.0.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Tests.scala]: SPARK-26491, commit ID: 1a641525e60039cc6b10816e946cb6f44b3e2696#diff-8b4ea8f3b0cc1e7ce7e943de1abbb165

File: core/src/main/scala/org/apache/spark/internal/config/Tests.scala [hunk 8]
```diff
@@ -22,43 +22,54 @@ private[spark] object Tests {
...
     .intConf
     .createWithDefault(4)
 
   val TEST_N_CORES_EXECUTOR = ConfigBuilder("spark.testing.nCoresPerExecutor")
+    .version("3.0.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Tests.scala]: SPARK-26491, commit ID: 1a641525e60039cc6b10816e946cb6f44b3e2696#diff-8b4ea8f3b0cc1e7ce7e943de1abbb165

File: core/src/main/scala/org/apache/spark/internal/config/Tests.scala [hunk 9]
```diff
@@ -22,43 +22,54 @@ private[spark] object Tests {
...
     .intConf
     .createWithDefault(2)
 
-  val RESOURCES_WARNING_TESTING =
-    ConfigBuilder("spark.resources.warnings.testing").booleanConf.createWithDefault(false)
+  val RESOURCES_WARNING_TESTING = ConfigBuilder("spark.resources.warnings.testing")
+    .version("3.1.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Tests.scala]: SPARK-29148, commit ID: 496f6ac86001d284cbfb7488a63dd3a168919c0f#diff-8b4ea8f3b0cc1e7ce7e943de1abbb165

File: core/src/main/scala/org/apache/spark/internal/config/Tests.scala [hunk 10]
```diff
@@ -22,43 +22,54 @@ private[spark] object Tests {
...
+    .booleanConf
+    .createWithDefault(false)
 
   val RESOURCE_PROFILE_MANAGER_TESTING =
     ConfigBuilder("spark.testing.resourceProfileManager")
+      .version("3.1.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Tests.scala]: SPARK-29148, commit ID: 496f6ac86001d284cbfb7488a63dd3a168919c0f#diff-8b4ea8f3b0cc1e7ce7e943de1abbb165

**Comment by HyukjinKwon** [comment]
@beliefer, let's merge this and https://github.com/apache/spark/pull/27757. It's unlikely to revert individually, and better to handle them in batch.

**Comment by beliefer** [comment]
@HyukjinKwon OK

**Comment by HyukjinKwon** [comment]
Thanks, @beliefer. Shall we also update PR title and description? We can close #27757 too.

**Comment by beliefer** [comment]
> Thanks, @beliefer. Shall we also update PR title and description? We can close #27757 too.

OK.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by HyukjinKwon** [comment]
Merged to branch-3.0 too

**Pull request closed without merging**